src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/QueryHelper.scala

```diff
@@ -65,11 +65,11 @@ private[querygeneration] case class QueryHelper(
         p.map(
           e =>
             colSet.find(c => c.exprId == e.exprId) match {
-              case Some(a) =>
+              case Some(a) if e.isInstanceOf[AttributeReference] =>
                 AttributeReference(a.name, a.dataType, a.nullable, a.metadata)(
```

Comment:
@sfc-gh-ema Do you remember why we do this here? I know we want to rename sub queries. But is it ok to only rename `AttributeReference` ?

Reply:
+1. @Mingli-Rui How confident are you that this is a safe change?